ui/src/app/views/rename-modal/component.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+
+import {
+  Button
+} from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import {ReactComponent} from 'app/utils';
+
+
+interface RenameModalProps {
+  resource: any;
+  resourceType: string;
+  onRename: Function;
+  onCancel: Function;
+}
+interface RenameModalState {
+  loading: boolean;
+  newName: string;
+}
+
+@ReactComponent({
+  selector: 'app-rename-modal',
+  propNames: ['resource', 'resourceType', 'onRename', 'onCancel']
+})
+export class RenameModalComponent extends React.Component<RenameModalProps, RenameModalState> {
+  constructor(props: RenameModalProps) {
+    super(props);
+    this.state = {
+      loading: false,
+      newName: ''
+    };
+  }
+
+  private emitRename(): void {
```

Comment:
nit: I realize this was copied from the previous code, but since we're not technically emitting an event any more, what about a simpler name such as `save` or `rename`?

Reply:
Agreed - done.